chiselFrontend/src/main/scala/chisel3/core/Bits.scala

```diff
@@ -67,6 +90,7 @@ sealed abstract class Bits(dirArg: Direction, width: Width, override val litArg:
     if (isLit()) {
       Bool(((litValue() >> x.toInt) & 1) == 1)
     } else {
+      Binding.checkSynthesizable(this, s"'this' ($this)")
```

Comment:
Less to update is better, more so if the stuff you need to update is concentrated in one place (like the FIRRTL emitter, as opposed to spread out through the Data and friends logic).

I see the problem with internal nodes now, though maybe that's also an emitter level issue?